src/bitcoin-wallet.cpp

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2016-2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#if defined(HAVE_CONFIG_H)
+#include <config/bitcoin-config.h>
+#endif
+
+#include <chainparams.h>
+#include <chainparamsbase.h>
+#include <consensus/consensus.h>
+#include <logging.h>
+#include <util.h>
+#include <utilstrencodings.h>
+#include <wallet/wallettool.h>
+
+#include <stdio.h>
+
+const std::function<std::string(const char*)> G_TRANSLATION_FUN = nullptr;
+
+static void SetupWalletToolArgs()
+{
+    SetupChainParamsBaseOptions();
+
+    gArgs.AddArg("-?", "This help message", false, OptionsCategory::OPTIONS);
+    gArgs.AddArg("-datadir=<dir>", "Specify data directory", false, OptionsCategory::OPTIONS);
+    gArgs.AddArg("-wallet=<wallet-name>", "Specify wallet name", false, OptionsCategory::OPTIONS);
+
+    gArgs.AddArg("info", "Get wallet info", false, OptionsCategory::COMMANDS);
+    gArgs.AddArg("create", "Create new wallet file", false, OptionsCategory::COMMANDS);
+    gArgs.AddArg("salvage", "Recover readable keypairs", false, OptionsCategory::COMMANDS);
+    gArgs.AddArg("zaptxs", "Remove all transactions including metadata (will keep keys)", false, OptionsCategory::COMMANDS);
+
+    // Hidden
+    gArgs.AddArg("-h", "", false, OptionsCategory::HIDDEN);
+    gArgs.AddArg("-help", "", false, OptionsCategory::HIDDEN);
+}
+
+static bool WalletAppInit(int argc, char* argv[])
+{
+    SetupWalletToolArgs();
+    std::string error;
+    if (!gArgs.ParseParameters(argc, argv, error)) {
+        fprintf(stderr, "Error parsing command line arguments: %s\n", error.c_str());
+        return false;
+    }
+    if (argc < 2 || HelpRequested(gArgs)) {
+        std::string usage = strprintf("%s bitcoin-wallet version", PACKAGE_NAME) + " " + FormatFullVersion() + "\n\n" +
+                                      "wallet-tool is an offline tool for creating and interacting with Bitcoin Core wallet files.\n" +
+                                      "By default wallet-tool will act on wallets in the default wallet directory in the datadir.\n" +
```

Comment:
Nit: "the datadir for mainnet" (since `testnet=1` in `bitcoin.conf` is ignored)

Reply:
fixed